js/algos.js

```diff
@@ -0,0 +1,125 @@
+      // and then using that index to access the string in the original array      
+// Output: A string
+function longestItem(array) {
+var lengths = []; 
+var max = 0;
+  array.forEach(function (item, index, array) {
+    lengths.push(item.length);
+  });
+  lengths.forEach(function (item, index, array) {
+    if (max < item) {
+      max = item;
+    } 
+  });
+return array[lengths.indexOf(max)];
+}
+ 
+//objectComparer function will take in two objects, and compare their key-value pairs to give. 
+// If there is a keys-value pair match, function will return true. if not, function will return false. 
+  // Input: Two objects 
+  // Steps: 
+    // To check for matching key-value pair, 
+      // Loop through object 1, 
+          // Check if the  value is the same both in object 1 and object 2, using the same key.
+          // Only returns true if there is a key-value match, because it uses the same key for both object values
+              // IF returns true 
+                 // function returns TRUE.              
+      //ELSE function returns FALSE.
+// OUTPUT: True / False                      
+function objectComparer(object1, object2) {
+  for(var key1 in object1) {
+    object2[key1] == object1[key1];
+      if (object2[key1] == object1[key1]){
+        return true; 
+      }   
+  }
+  return false; 
+ }
+ 
+//For the function that generates a random word array of the given length, Im splitting into two functions:
+// One function (randomWord) generates the random word thats from 1 to 10 letters long. 
+// Second function (wordArray) generates a random words array of the specified length. 
+  // randomWord function 
+  // Input: none
+    // Steps:
+    // declare alphabet variable
+    // declare variable that will select the length of the word randomly from 1 to 10. 
+    // declare empty string to later store the new word. 
+      // Loop for the amount of the randomly selected length. 
+        // generate random number from 0 to max the alphabet length, not included. 
+        // Select a random letter by using that random number as index to the alphabet string. 
+        // Add that letter to empty string.  
+    //return string
+  //OUTPUT: string.  
+
+  // wordArray function
+  // Input: number
+    // Steps: 
+    // declare empty array
+    // Loop for the amount passed as parameter
+      // call randomWord to generate a random word
+      // push that random word to empty array
+  //OUTPUT: array. 
+function randomWord() {
+var chars = "abcdefghijklmnopqrstuvwxyz";
+var randomLength = Math.floor(Math.random() * 10) + 1; 
+var string= '';
+   for (var letters = 0; letters < randomLength; letters ++) {
+      var randomLetter= chars[Math.floor(Math.random() * (chars.length))];
+      string += randomLetter; 
+   }
+return string; 
+}
+function wordArray(num){
+var words = []; 
+  for (var times = 0; times < num; times ++){
+    randomWord(); 
```

Comment:
I think this is good because you are splitting up the complexity and encapsulating your processes for reuse and easy refactor